### PR TITLE
fix: correctness sweep across record, snapshot, report, loader, assert, and run

### PR DESCRIPTION
**Pull request by nao1215**
A correctness pass across `record`, `snapshot`, the report formats, the loader, the assertion matchers, secret masking, and the run engine. Each defect was surfaced by an edge-case bug hunt and fixed with a reproduction test first. No new features; twenty-one distinct defects across seven packages, grouped into the changelog's Fixed entries.

record

The `record --pty` anchor stripped ANSI from the visible line and joined the result, so a colored prompt produced an `expect`/`stdout: contains` value with mid-line color codes removed that was never a substring of the raw pty stdout the replay matches — the generated spec failed on replay. `stableLine` now returns the longest color-free run of the last visible line, which is verbatim in the raw output. Separately, raw-mode capture delivers one keystroke per read and consecutive input bursts were not coalesced, so a six-character password generated seven single-character `${env:ATAGO_SECRET_n}` sends; `AppendInput` now merges consecutive bursts with the same echo state.

snapshot

A dir-tree manifest joins a symlink's name and target with ` -> ` but did not escape it inside either field, so a symlink named `a -> b` pointing at `c` rendered the same line as one named `a` pointing at `b -> c` and falsely matched its golden. Snapshot normalization also masked the home and workdir prefixes with a bare substring replace, so `/home/nao` corrupted `/home/naoki` into `~ki`, a container `HOME=/` turned every absolute path into tildes, and workdir `/tmp/run1` turned `/tmp/run10` into `<workdir>0`. Masking is now path-component aware, skips a root home, and runs after the uuid/timestamp/port maskers so it stays idempotent.

report

A unified-diff hunk with a zero-line side emitted `@@ -1,0 ...` instead of the GNU `@@ -0,0 ...` that patch(1) relies on, and an empty side carried a spurious no-newline marker. A `--report tap` diagnostic message was escaped for the bare ok/not-ok line and then quoted, so `issue #42` reached the consumer as `issue \#42`. A `--report junit` `time` attribute serialized a sub-millisecond duration as `1.5e-06`, which the JUnit/Surefire schema rejects. The console summary headline summed each suite's own duration, overcounting when `--parallel` runs suites concurrently.

run engine

`--fail-fast` only stopped scenarios within one suite, so a failing first spec still let every later spec run; scheduling now halts across files. `--tag` and `--skip-tag` were single-string flags that kept only the last occurrence, so `--tag a --tag b` silently dropped `a`; both are now repeatable and OR their values like `--filter`.

assert and security

The `gt`/`gte`/`lt`/`lte` matchers rounded the selected value through float64, so a JSON integer beyond 2^53 compared equal to its neighbor; they now use exact big.Int arithmetic like `equals`. A scalar `equals` fallback compared operands by their printed form, so a JSON boolean `true` matched `equals: "true"`. Secret masking let one secret consume bytes an overlapping secret needed, leaking its tail in cleartext; every occurrence is now masked in one pass over the original text.

loader

A `mock_server` step in a scenario's `steps` or `teardown` was silently accepted though it is `suite.setup`-only. An unknown key in the `exit_code` mapping form bypassed strict decoding. A `dir.glob` pattern was never compile-checked at load time. A negative `timeout`/`delay` on the suite, run, defaults, runner, service readiness, and mock-route fields was accepted and produced an already-expired context.

Verification

`make test`, `make lint`, and `make e2e` (228 scenarios) all pass. Every fix carries a reproduction test that fails before the change; the `--report junit` time-format change regenerated `site/samples/report.junit.xml` via `make site`.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved report formatting for diffs, TAP, JUnit, and console summaries.
  * Fixed snapshot/path masking so similar paths and home directories are handled correctly.
  * Made secret masking more reliable, including overlapping secrets and bursty hidden input.
  * Improved replay anchors for terminal recordings so prompts and colored output replay more consistently.
  * Tightened validation for durations, glob patterns, exit-code rules, and mock server placement.
  * Updated tag filtering and fail-fast behavior so runs stop and select scenarios more predictably.

### CODE REVIEW

**Comment by nao1215** [comment]
Thanks for the review — all points addressed in c2c1864:

- masker.go: advance one byte after a hit so self-overlapping repeats of a secret are fully masked (`aaaa` in `aaaaaa`), with a regression test.
- snapshot.go: only a separator or whitespace/control byte is a masking boundary now, so a punctuation-suffixed sibling like `/tmp/run1+cache` is left intact; added a test.
- diff.go: the no-newline marker is judged per side, so two sides that both lack a trailing newline are both annotated; added a both-no-newline regression.
- pty.go: the record ANSI pattern now matches colon SGR subparameters (truecolor), so they no longer leak into an anchor.
- run.go: an interrupt that skips every suite before any result now exits non-zero instead of 0; fixed the `cancelled` spelling.
